### PR TITLE
refactor: optimize comment and reply logic to reduce duplicate code

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind improvement
/area core
/milestone 2.19.x

#### What this PR does / why we need it:
优化评论和回复的逻辑并减少重复代码

#### Does this PR introduce a user-facing change?
```release-note
None
```

### CODE REVIEW

File: application/src/main/java/run/halo/app/content/comment/AbstractCommentService.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    private Mono<CommentStats> fetchStats(String meterName) {
+        Assert.notNull(meterName, "The reply must not be null.");
+        return counterService.getByName(meterName)
+            .map(counter -> CommentStats.builder()
+                .upvote(counter.getUpvote())
+                .build()
+            )
+            .defaultIfEmpty(CommentStats.empty());
```
Comment (JohnNiang) [inline application/src/main/java/run/halo/app/content/comment/AbstractCommentService.java:76]: 建议用 `switchIfEmpty(CommentStats::empty)`。

File: application/src/main/java/run/halo/app/content/comment/AbstractCommentService.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    protected Mono<OwnerInfo> getOwnerInfo(Comment.CommentOwner owner) {
+        if (User.KIND.equals(owner.getKind())) {
+            return userService.getUserOrGhost(owner.getName())
+                .map(OwnerInfo::from);
+        }
+        if (Comment.CommentOwner.KIND_EMAIL.equals(owner.getKind())) {
+            return Mono.just(OwnerInfo.from(owner));
+        }
+        throw new IllegalStateException(
```
Comment (JohnNiang) [inline application/src/main/java/run/halo/app/content/comment/AbstractCommentService.java:57]: 建议返回 Mono.error，而不是直接抛异常。

File: application/src/main/java/run/halo/app/content/comment/AbstractCommentService.java (hunk 1 above)
Comment (guqing) [reply application/src/main/java/run/halo/app/content/comment/AbstractCommentService.java]: Done

File: application/src/main/java/run/halo/app/content/comment/AbstractCommentService.java (hunk 2 above)
Comment (guqing) [reply application/src/main/java/run/halo/app/content/comment/AbstractCommentService.java]: good catch

**Review by JohnNiang — COMMENTED** [review]
/approve

It works very well.

**Review by JohnNiang — APPROVED** [review]

**Review by ruibaby — APPROVED** [review]
/lgtm

**Pull request merged**